src/main/java/org/junit/experimental/theories/internal/AllMembersSupplier.java

```diff
@@ -129,16 +132,37 @@ else if (Iterable.class.isAssignableFrom(type)) {
         }
     }
 
-    private void addArrayValues(ParameterSignature sig, String name, List<PotentialAssignment> list, Object array) {
+    // Nullable array from addDataPointsValues(Class<?> type, ParameterSignature sig, String name,
+    //            List<PotentialAssignment> list, Object value)
+    private void addArrayValues(ParameterSignature sig, String name, List<PotentialAssignment> list, @Nullable Object array) {
         for (int i = 0; i < Array.getLength(array); i++) {
             Object value = Array.get(array, i);
             if (sig.canAcceptValue(value)) {
                 list.add(PotentialAssignment.forValue(name + "[" + i + "]", value));
             }
         }
     }
-    
-    private void addIterableValues(ParameterSignature sig, String name, List<PotentialAssignment> list, Iterable<?> iterable) {
+
+    // Nullable array from addDataPointsValues(Class<?> type, ParameterSignature sig, String name,
+    //            List<PotentialAssignment> list, Object value)
+    private void addIterableValues(ParameterSignature sig, String name, List<PotentialAssignment> list, @Nullable Iterable<?> iterable) {
+        // [dereference.of.nullable] TRUE_POSITIVE
+        // dereference of possibly-null reference iterable
+        // the caller method only ensures the type is Iterable.class
+        // but did not ensure that value is non-null
+        //
+        // One example to raise NPEs is:
+        // @RunWith(Theories.class)
+        // public class Empty {
+        //    @DataPoints
+        //    public static List<String> dataPoints = null;
+        //    @Theory
+        //    public void testList(List<String> list) {
+        //        for (String s: list) {
+        //            System.out.println(s);
+        //        }
+        //    }
+        // }
         Iterator<?> iterator = iterable.iterator();
```

Comment:
See and run eval_test/TestAllMembersSuppliers.java, it raises an NPE at the place you mentioned should be false positive above.

Reply:
You mentioned the fields can "only be instantiated once, when the class is loaded", I am not sure where did you see it. From the section 8.3.2 you pointed out, the sentence might related to your comment is 
>If the declarator is for a class variable (that is, a static field), then the variable initializer is evaluated and the assignment performed exactly once, when the class is initialized (§12.4.2). 

but it doesn't support your idea (plus your example in helpful link is primitive type which must be initialized), and JUnit 4 document leaves out that data annotated with `@DataPoints` will only be used in corresponding `@Theories` methods once.

You mentioned two possibilities, but your example are just two aliases of assigning the same static field. Another possibility should be:
>@DataPoints
 public static String[] generatedDataPoints() {
     return new String[] { ... };
 }

I agree with your point that developers of JUnit4 `@DataPoints` is intended for the initialized fields with or methods that return some iterable-typed data. But the document only mentions about the data's type, not its underlying implementation. Neither the document mentions data cannot be null.

Reply:
8.3.2. mentions static field can only be assigned once.
Then, if you assign it with null, it cannot be assigned to an array or iterable-typed field.

The example you gave is just the second case of using @Datapoints, which is for a method
>we could only annotate an array or iterable-typed field or *method* with @DataPoints.

But I'm just talking about the first case, a static field.

Reply:
Since the implementation doesn't check whether the underlying value is null or not, and the documentation doesn't explicitly say that users shouldn't pass null to the static iterable-typed field, although `@DataPoints` is probably designed for users to pass non-null value. We decided it's still TRUE_POSITIVE.